widgets/search-box/search-box.js

```diff
@@ -100,13 +101,33 @@ function searchBox({
 
       // Add all the needed attributes and listeners to the input
       this.addDefaultAttributesToInput(input, initialState.query);
+
+      // Keep keyup to handle searchOnEnterKeyPressOnly
       input.addEventListener('keyup', (e) => {
         helper.setQuery(input.value);
-        if (!searchOnEnterKeyPressOnly || e.keyCode === KEY_ENTER) {
+        if (searchOnEnterKeyPressOnly && e.keyCode === KEY_ENTER) {
+          helper.search();
+        }
+
+        if (window.attachEvent && e.keyCode === KEY_SUPPRESS) {
           helper.search();
         }
       });
 
+      function inputCallback(e) {
+        let target = (e.currentTarget) ? e.currentTarget : e.srcElement;
+        helper.setQuery(target.value);
+        if (!searchOnEnterKeyPressOnly) {
+          helper.search();
+        }
+      }
+
+      if (window.attachEvent) {
+        input.attachEvent('onpropertychange', inputCallback);
```

Comment:
Same here, can you add a comment stating this is only for IE8/9, they only support onpropertychange not the input event.